lms/djangoapps/grades/new/course_grade.py

```diff
@@ -31,11 +31,15 @@ class CourseGrade(object):
     def __init__(self, student, course, course_structure):
         self.student = student
         self.course = course
-        self.course_version = getattr(course, 'course_version', None)
-        self.course_edited_timestamp = getattr(course, 'subtree_edited_on', None)
-        self.course_structure = course_structure
         self._percent = None
         self._letter_grade = None
+
+        self.course_structure = course_structure
+        if self.course_structure:
```

Comment:
Why the different approach here of going deeper into the `self.course_structure` object?  Does this logic all break and cause a bug again if `self.course_structure` isn't present?